pkg/model/openstackmodel/servergroup.go

```diff
@@ -159,28 +159,28 @@ func (b *ServerGroupModelBuilder) buildInstances(c *fi.ModelBuilderContext, sg *
 			case kops.InstanceGroupRoleBastion:
 				t := &openstacktasks.FloatingIP{
 					Name:      fi.String(fmt.Sprintf("%s-%s", "fip", *instanceTask.Name)),
-					Server:    instanceTask,
 					Lifecycle: b.Lifecycle,
 				}
 				c.AddTask(t)
+				instanceTask.FloatingIP = t
 			case kops.InstanceGroupRoleMaster:
 				if b.Cluster.Spec.CloudConfig.Openstack.Loadbalancer == nil {
 					t := &openstacktasks.FloatingIP{
 						Name:      fi.String(fmt.Sprintf("%s-%s", "fip", *instanceTask.Name)),
-						Server:    instanceTask,
 						Lifecycle: b.Lifecycle,
 					}
 					c.AddTask(t)
 					b.associateFIPToKeypair(t)
+					instanceTask.FloatingIP = t
 				}
 			default:
-				if !b.UsesSSHBastion() {
+				if b.Cluster.Spec.Topology.Nodes == "public" {
```

Comment:
is this change backwards compatible to old clusters?

Reply:
The reason I added this was that I got floating IPs on all nodes otherwise. Even with private topology.
There is a lot of validation missing for configuring an openstack cluster, where e.g private topology on masters + floating ips on masters should be conflicting. In addition there is `ig.spec.associatePublicIP` that comes into play too, where floating IPs should be added regardless.

Happy to amend this any other way.